packages/gateway/src/middleware/assign-request-id.ts

```diff
@@ -0,0 +1,16 @@
+import { RequestHandler } from 'express';
+import uuid from 'uuid';
+import { metricEmitter, MetricsChannels } from '../metric';
+
+const assignRequestId: RequestHandler = (req, res, next) => {
+  req.id = uuid.v4();
+  metricEmitter.emit(MetricsChannels.GOT_REQUEST, req.id, {
+    query: req.body,
+    userAgent: req.headers['user-agent'],
+    ip: req.ip,
```

Comment:
`req.body` and `req.headers` are optional and might be undefined, `req.headers` may not contain _user-agent_. Shall we check for this and send empty strings in that case?